doc/esmvalcore/preprocessor.rst

```diff
@@ -65,27 +99,106 @@ the derivation can be forced by setting the appropriate flag.
       force_derivation: false
 
 The required arguments for this module are two boolean switches:
-* derive: activate variable derivation
-* force_derivation: force variable derivation even if the variable is
-directly available in the input data.
+
+* ``derive``: activate variable derivation
+* ``force_derivation``: force variable derivation even if the variable is
+  directly available in the input data.
 
 See also :func:`esmvalcore.preprocessor.derive`.
 
 
-CMOR check and dataset-specific fixes
+.. _CMOR check and dataset-specific fixes:
+
+CMORization and dataset-specific fixes
 ======================================
 .. warning::
-   Documentation of _reformat.py, check.py and fix.py to be added
 
+   Section to be added.
+
+
+.. _Vertical interpolation:
 
 Vertical interpolation
 ======================
-.. warning::
-   Documentation of  _regrid.py (part 1) to be added
+Vertical level selection is an important aspect of data preprocessing since it
+allows the scientist to perform a number of metrics specific to certain levels
+(whether it be air pressure or depth, e.g. the Quasi-Biennial-Oscillation (QBO)
+u30 is computed at 30 hPa). Dataset native vertical grids may not come with the
+desired set of levels, so an interpolation operation will be needed to regrid
+the data vertically. ESMValTool can perform this vertical interpolation via the
+``extract_levels`` preprocessor. Level extraction may be done in a number of
+ways.
+
+Level extraction can be done at specific values passed to ``extract_levels`` as
+``levels:`` with its value a list of levels (note that the units are
+CMOR-standard, Pascals (Pa)):
+
+.. code-block:: yaml
 
+    preprocessors:
+      preproc_select_levels_from_list:
+        extract_levels:
+          levels: [100000., 50000., 3000., 1000.]
+          scheme: linear
+
+It is also possible to extract the CMIP-specific, CMOR levels as they appear in
+the CMOR table, e.g. ``plev10`` or ``plev17`` or ``plev19`` etc:
 
-Land/Sea/Ice masking
-====================
+.. code-block:: yaml
+
+    preprocessors:
+      preproc_select_levels_from_cmip_table:
+        extract_levels:
+          levels: {cmor_table: CMIP6, coordinate: plev10}
+          scheme: nearest
+
+Of good use is also the level extraction with values specific to a certain
+dataset, without the user actually polling the dataset of interest to find out
+the specific levels: e.g. in the example below we offer two alternatives to
+extract the levels and vertically regrid onto the vertical levels of
+``ERA-Interim``:
+
+.. code-block:: yaml
+
+    preprocessors:
+      preproc_select_levels_from_dataset:
+        extract_levels:
+          levels: ERA-Interim
+          # This also works, but allows specifying the pressure coordinate name
+          # levels: {dataset: ERA-Interim, coordinate: air_pressure}
+          scheme: linear_horizontal_extrapolate_vertical
```

Comment:
Is there a description of the available schemes somewhere?

Reply:
Yes, in the regridding subsection of this section.